### ISSUE TITLE
Option to for path prefix for info-json file (e.g. save to subdirectory)

### ISSUE THREAD

**Issue by mk-pmb** [ytdl-org/youtube-dl#30847]
Option to for path prefix for info-json file (e.g. save to subdirectory)

## Checklist

- [x] I'm reporting a feature request
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've searched the bugtracker for similar feature requests including closed ones

## Description

I like to use `--write-info-json`, and I also almost like how it uses the filename template from the `--output` option. However, it's just "almost" because I want it in a subdirectory.

Future me might want it in an entirely different path, or with a filename prefix in the same directory, so I prefer the more general option of a path prefix.

If the prefix causes the info json file path to contain directory information, I'd like yt-dl to make a weak attempt at creating that directory. Failure in this tshall be silent and shall be ignored. The thing I actually care about is whether the info json can be written, and on exotic file systems, that might be entirely independent from directory creation.

In case this feature exists already, the feature request is instead about adding a hint to `--help`'s description of `--write-info-json` on how to have it be written to a subdirectory.

Would maintainers like to merge this? Shall I try and implement it?

**Comment by dirkf** [comment]
Although this could be done, you could also add a `--exec ...` option with a shell command to move the JSON file to your desired subdirectory.

**Comment by mk-pmb** [comment]
Thanks for the idea. My gut feeling prefers the path prefix though.

**Comment by mk-pmb** [comment]
Indeed, when I had a closer look, I found the annotations file etc. That convinced me, for now. Let's postpone it until someone comes up with an elegant solution for the other auxiliary files as well.

**Comment by pukkandan** [comment]
yt-dlp allows using `infojson:` prefix in `-o`. Eg:  `-o "infojson:D:\metadata\%(id)s.%(ext)s" -o "D:\videos\%(title)s.%(ext)s"`. Similar syntax could be implemented here. The advantage is that the single option `--output` handles setting templates for the video, thumbnail, infojson etc.

PS: When trying to implement this for thumbnail/subtitles, take care not to break embedding

**Comment by dirkf** [comment]
I suspect this is either an error-prone addition or a tricky back-port. A user who really needs it has the options of `--exec ...`, external scripting, embedding, or just using yt-dlp.
